libraries/SD_MMC/src/SD_MMC.cpp

```diff
@@ -12,29 +12,76 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include "pins_arduino.h"
 #include "SD_MMC.h"
-#if !defined(CONFIG_IDF_TARGET_ESP32S2) && !defined(CONFIG_IDF_TARGET_ESP32C3) //SDMMC does not work on ESP32S2
+#ifdef SOC_SDMMC_HOST_SUPPORTED
 #include "vfs_api.h"
 
-extern "C" {
-#include <sys/unistd.h>
-#include <sys/stat.h>
 #include <dirent.h>
 #include "esp_vfs_fat.h"
 #include "driver/sdmmc_host.h"
 #include "driver/sdmmc_defs.h"
 #include "sdmmc_cmd.h"
-}
+#include "soc/sdmmc_pins.h"
 #include "ff.h"
 
 using namespace fs;
-/*
 
-*/
 
 SDMMCFS::SDMMCFS(FSImplPtr impl)
     : FS(impl), _card(NULL)
-{}
+{
+#ifdef SOC_SDMMC_USE_GPIO_MATRIX
+    _pin_clk = 1;
+    _pin_cmd = 2;
+    _pin_d0 = 3;
+    _pin_d0 = 4;
+    _pin_d0 = 5;
+    _pin_d0 = 6;
```

Comment:
These will ideally have to come from pins_arduino.h, if the board variant specifies the pins.

Reply:
if not defined in pins_arduino.h, they should be `GPIO_NUM_NC` by default and the driver should not start if it has missing pins. That way we handle both boards with predefined pins and custom user configs.

Reply:
maybe something like:
```c
#ifdef BOARD_HAS_SDMMC
    _pin_clk = SDMMC_CLK;
    _pin_cmd = SDMMC_CMD;
    _pin_d0 = SDMMC_D0;
  #ifdef BOARD_HAS_1BIT_SDMMC
    _pin_d1 = -1;
    _pin_d2 = -1;
    _pin_d3 = -1;
  #else
    _pin_d1 = SDMMC_D1;
    _pin_d2 = SDMMC_D2;
    _pin_d3 = SDMMC_D3;
  #endif
#elif CONFIG_IDF_TARGET_ESP32
    _pin_clk = SDMMC_SLOT1_IOMUX_PIN_NUM_CLK;
    _pin_cmd = SDMMC_SLOT1_IOMUX_PIN_NUM_CMD;
    _pin_d0 = SDMMC_SLOT1_IOMUX_PIN_NUM_D0;
    _pin_d1 = SDMMC_SLOT1_IOMUX_PIN_NUM_D1;
    _pin_d2 = SDMMC_SLOT1_IOMUX_PIN_NUM_D2;
    _pin_d3 = SDMMC_SLOT1_IOMUX_PIN_NUM_D3;
#else
    _pin_clk = -1;
    _pin_cmd = -1;
    _pin_d0 = -1;
    _pin_d1 = -1;
    _pin_d2 = -1;
    _pin_d3 = -1;
#endif
```